test/throttle_SUITE.erl

```diff
@@ -0,0 +1,214 @@
+    ?assertMatch({error, cannot_change_rate},
+                 amoc_throttle:change_rate(?FUNCTION_NAME, 100, ?DEFAULT_INTERVAL + 1)).
+
+send_and_wait(_) ->
+    %% it failts if the throttle wasn't started yet
+    ?assertMatch({error, no_throttle_process_registered},
+                 amoc_throttle:send_and_wait(?FUNCTION_NAME, receive_this)),
+    %% Start 100-per-10ms throttle with a single process
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 100, 10, 1)),
+    %% send_and_wait passes fine
+    ?assertMatch(ok, amoc_throttle:send_and_wait(?FUNCTION_NAME, receive_this)),
+    %% One message is received sufficiently fast
+    amoc_throttle:send(?FUNCTION_NAME, receive_this),
+    ?assertMatch(ok, receive_msg_in_timeout(receive_this, 100)),
+    %% If someone else fills the throttle heavily,
+    %% it will take proportionally so long to execute for me
+    fill_throttle(?FUNCTION_NAME, 100 * 10),
+    amoc_throttle:send(?FUNCTION_NAME, receive_this),
+    ?assertMatch({error, not_received_yet}, receive_msg_in_timeout(receive_this, 200)).
+
+run_with_interval_zero_limits_only_number_of_parallel_executions(_) ->
+    %% Start 10 actions at once in 10 processes
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 10, 0, 1)),
+    %% If someone else fills the throttle heavily,
+    %% it will take proportionally so long to execute for me
+    fill_throttle(?FUNCTION_NAME, 100),
+    amoc_throttle:send(?FUNCTION_NAME, receive_this),
+    ?assertMatch(ok, receive_msg_in_timeout(receive_this, 200)).
+
+pause_and_resume(_) ->
+    %% Start 100-per-10ms throttle with a single process
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 100, 10, 1)),
+    %% send_and_wait passes fine
+    ?assertMatch(ok, amoc_throttle:send_and_wait(?FUNCTION_NAME, receive_this)),
+    %% pauses runs correctly
+    ?assertMatch(ok, amoc_throttle:pause(?FUNCTION_NAME)),
+    %% It is paused, so messages aren't received
+    amoc_throttle:send(?FUNCTION_NAME, receive_this),
+    ?assertMatch({error, not_received_yet}, receive_msg_in_timeout(receive_this, 200)),
+    %% After resume the message is then received
+    ?assertMatch(ok, amoc_throttle:resume(?FUNCTION_NAME)),
+    ?assertMatch(ok, receive_msg_in_timeout(receive_this, 200)).
+
+get_state(_) ->
+    ?assertMatch({ok, started}, amoc_throttle:start(?FUNCTION_NAME, 100, 60000, 1)),
+    State = get_state_of_one_process(?FUNCTION_NAME),
+    ?assertMatch(#{name := ?FUNCTION_NAME,
+                   interval := 60000,
+                   delay_between_executions := 600},
+                 State).
+
+
+%% Helpers
+assert_telemetry_event(Name, Count, Throttle, Rate, Interval) ->
+    TelemetryEvents = telemetry_helpers:get_calls([amoc, throttle]),
+    LowRateEvent = fun({EventName, Measurements, Metadata}) ->
+                           Name =:= EventName andalso
+                           1 =:= maps:get(Count, Measurements, 1) andalso
```

Comment:
are we checking here equality to the default value?

Reply:
We are checking that the key exists in the map, which wtf there's a `maps:is_key/2`, sorry xD